### ISSUE TITLE
Make `clamp` function internal

### ISSUE THREAD

**Issue by SerVB** [korlibs/kmem#8]
In KorGE, there are two functions available.

![image](https://user-images.githubusercontent.com/20105593/103457712-af41b080-4d12-11eb-9ce8-583024125842.png)

As [discussed](https://discord.com/channels/728582275884908604/728616376910217258/794909177700548618), the function in kmem should probably be internal

**Comment by SerVB** [comment]
Just checked and see that the one from kmem is used in many other korlibs:
![image](https://user-images.githubusercontent.com/20105593/103458457-7fe27200-4d19-11eb-85d1-eba2d6efbd5e.png)

So need to change them too...

The one from korma isn't used anywhere for now.

**Comment by soywiz** [comment]
Could be provided on each library as an internal inline function using `.coerceIn` as implementation https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/coerce-in.html

**Comment by SerVB** [comment]
Hey @soywiz, what do you think about deprecating `clamp` in favor of `coerceIn`? I just think it's not worth it to have a library method when the same is available in stdlib for all the targets (since Kotlin 1.3). FMPOV, it's better to have only one obvious way to do the single thing (as in Zen of Python 😁). I could make a PR with deprecations and replacements.

**Comment by SerVB** [comment]
The only drawback I see is that `coerceIn` has additional check:

`if (minimumValue > maximumValue) throw IllegalArgumentException("Cannot coerce value to an empty range: maximum $maximumValue is less than minimum $minimumValue.")`

So it will be incompatible change.

**Comment by soywiz** [comment]
Uhm. That could be an option. I usually like `clamp` naming because it is somehow standard and that name comes up in my mind instead of `corceIn`:

* https://en.wikipedia.org/wiki/Clamping_(graphics)
* https://developer.mozilla.org/en-US/docs/Web/CSS/clamp()
* https://en.cppreference.com/w/cpp/algorithm/clamp
* https://docs.microsoft.com/es-es/dotnet/api/system.math.clamp?view=net-5.0
* https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/clamp.xhtml

**Comment by SerVB** [comment]
Yes, of course `clamp` naming is more standard. So let me modify the suggestion a bit: maybe if we decide to deprecate `clamp`, we shouldn't remove it completely. It can be left as deprecated with ERROR level for like forever. Thus it will be searchable in IDE but can't be used. Its ReplaceWith will suggest using `coerceIn`. WDYT?

**Comment by soywiz** [comment]
Okay, that could work. Let's start with marking it as deprecated with a warning?

**Comment by SerVB** [comment]
OMG, I've started marking and replacing and it's quite a big task. I have a crazy idea of renaming stdlib function to the standard name, since Kotlin now has fast iteration cycle. I've filed a ticket [KT-46381](https://youtrack.jetbrains.com/issue/KT-46381), feel free to comment/vote.

Anyway, I think it's worth making `clamp` more consistent in korlibs, so I will continue marking and replacing and will submit a PR later.
